From d8b51cd1a0a080759eb66fe04814fd174c8afb42 Mon Sep 17 00:00:00 2001
From: Franz-Josef Haider <frajo@frajo.fi>
Date: Tue, 30 Jun 2026 21:20:46 +0300
Subject: [PATCH 3/9] Revert "Always pass useHostHwBinder to modified methods
 that can accept it"

This reverts commit 4baa21bfab0405a10e9df4fb713b079f5665827d.
---
 transport/HidlBinderSupport.cpp | 19 +++++++------------
 transport/ServiceManagement.cpp |  5 ++---
 2 files changed, 9 insertions(+), 15 deletions(-)

diff --git a/transport/HidlBinderSupport.cpp b/transport/HidlBinderSupport.cpp
index a739056..3d81749 100644
--- a/transport/HidlBinderSupport.cpp
+++ b/transport/HidlBinderSupport.cpp
@@ -278,14 +278,12 @@ bool doesSupportHostBinder() {
 static bool gThreadPoolConfigured = false;
 
 void configureBinderRpcThreadpool(size_t maxThreads, bool callerWillJoin) {
-    bool useHostHwBinder = false;
-    status_t ret = ProcessState::self(useHostHwBinder)->setThreadPoolConfiguration(
+    status_t ret = ProcessState::self()->setThreadPoolConfiguration(
         maxThreads, callerWillJoin /*callerJoinsPool*/);
     LOG_ALWAYS_FATAL_IF(ret != OK, "Could not setThreadPoolConfiguration: %d", ret);
 
     if (doesSupportHostBinder()){
-        useHostHwBinder = true;
-        status_t ret = ProcessState::self(useHostHwBinder)->setThreadPoolConfiguration(
+        status_t ret = ProcessState::self(true)->setThreadPoolConfiguration(
             maxThreads, callerWillJoin /*callerJoinsPool*/);
         LOG_ALWAYS_FATAL_IF(ret != OK, "Could not setThreadPoolConfiguration for Host: %d", ret);
     }
@@ -296,16 +294,15 @@ void configureBinderRpcThreadpool(size_t maxThreads, bool callerWillJoin) {
 void joinBinderRpcThreadpool() {
     LOG_ALWAYS_FATAL_IF(!gThreadPoolConfigured,
                         "HIDL joinRpcThreadpool without calling configureRpcThreadPool.");
-    IPCThreadState::self(false)->joinThreadPool();
+    IPCThreadState::self()->joinThreadPool();
     if (doesSupportHostBinder()){
-        IPCThreadState::self(true)->joinThreadPool();
+        IPCThreadState::selfForHost()->joinThreadPool();
     }
 }
 
 int setupBinderPolling() {
     int fd;
-    const bool useHostHwBinder = false;
-    int err = IPCThreadState::self(useHostHwBinder)->setupPolling(&fd);
+    int err = IPCThreadState::self()->setupPolling(&fd);
 
     LOG_ALWAYS_FATAL_IF(err != OK, "Failed to setup binder polling: %d (%s)", err, strerror(err));
 
@@ -313,13 +310,11 @@ int setupBinderPolling() {
 }
 
 status_t handleBinderPoll() {
-    const bool useHostHwBinder = false;
-    return IPCThreadState::self(useHostHwBinder)->handlePolledCommands();
+    return IPCThreadState::self()->handlePolledCommands();
 }
 
 void addPostCommandTask(const std::function<void(void)> task) {
-    const bool useHostHwBinder = false;
-    IPCThreadState::self(useHostHwBinder)->addPostCommandTask(task);
+    IPCThreadState::self()->addPostCommandTask(task);
 }
 
 }  // namespace hardware
diff --git a/transport/ServiceManagement.cpp b/transport/ServiceManagement.cpp
index 9884c70..8612601 100644
--- a/transport/ServiceManagement.cpp
+++ b/transport/ServiceManagement.cpp
@@ -574,8 +574,7 @@ struct Waiter : IServiceNotification {
         // that thread, it will block forever because we hung up the one and only
         // binder thread on a condition variable that can only be notified by an
         // incoming binder call.
-        const bool useHostHwBinder = false;
-        if (IPCThreadState::self(useHostHwBinder)->isOnlyBinderThread()) {
+        if (IPCThreadState::self()->isOnlyBinderThread()) {
             LOG(WARNING) << "Can't efficiently wait for " << mInterfaceName << "/"
                          << mInstanceName << ", because we are called from "
                          << "the only binder thread in this process.";
@@ -767,7 +766,7 @@ sp<::android::hidl::base::V1_0::IBase> getRawServiceInternal(const std::string&
         if (doesSupportHostBinder())
         {
             if (transportRet == Transport::EMPTY) {
-                sm = defaultServiceManager1_1(/*useHostHwBinder*/true);
+                sm = defaultServiceManager1_1(/*useHostBinder*/true);
                 if (sm == nullptr) {
                     ALOGE("getService: defaultServiceManager() of host is null");
                     return nullptr;
-- 
2.54.0

